### PR TITLE
fix: allow last field of an object to be obfuscated

### PR DESCRIPTION
**Pull request by Nebulis**

### CODE REVIEW

File: test/e2e.v2.test.ts [hunk 1]
```diff
@@ -62,4 +63,58 @@ describe("v2 E2E Test Scenarios", () => {
...
+
+    test("obfuscate data when the obfuscated field is the last element of an object", async () => {
+      const newDocument = wrapDocument(
+        { key1: "value1", key2: { key22: "value22" }, ...openAttestationDatav2 },
+        { version: SchemaId.v2 }
+      );
+      const firstObfuscatedDocument = obfuscateDocument(newDocument, ["key2.key22"]);
+      expect(verifySignature(firstObfuscatedDocument)).toBe(true);
+      expect(validateSchema(firstObfuscatedDocument)).toBe(true);
+      expect(firstObfuscatedDocument.data.key2).toStrictEqual({});
```
Comment (gjj) [inline test/e2e.v2.test.ts:108]: this

Comment (Nebulis) [reply test/e2e.v2.test.ts:108]: Uncaught ReferenceError: this is not defined

Comment (gjj) [reply test/e2e.v2.test.ts:108]: I just saw this LOL 🤣

**Pull request merged**